src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,67 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+type Props = {
+  qwery: string
+  setQwery: (qwery: string) => void
+  selectedFilter: string
+  setSelectedFilter: (selectedFilter: string) => void
+};
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+export const TodoFilter: React.FC<Props> = (
+  {
+    qwery, setQwery, setSelectedFilter, selectedFilter,
+  },
+) => {
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQwery(event.target.value);
+  };
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+  const handleSelectedChange = (event: { target: { value: string } }) => {
+    setSelectedFilter(event.target.value);
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={selectedFilter}
+            onChange={handleSelectedChange}
+          >
+            <option value="all">All</option>
+            <option value="active">Active</option>
+            <option value="completed">Completed</option>
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={qwery}
+          onChange={handleInputChange}
         />
-      </span>
-    </p>
-  </form>
-);
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        {
+          qwery && (
+            <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                data-cy="clearSearchButton"
+                type="button"
+                className="delete"
+                onClick={() => setQwery('')}
```

Comment:
```suggestion
              <button
                data-cy="clearSearchButton"
                type="button"
                className="delete"
                onClick={() => setQwery('')}
                aria-label="delete query"
```
aria-label attribute fix eslin issue